x-pack/legacy/plugins/infra/public/containers/logs/log_highlights/log_summary_highlights.ts

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { useEffect, useMemo, useState } from 'react';
+import { debounce } from 'lodash';
+
+import { LogSummaryHighlightsQuery } from '../../../graphql/types';
+import { DependencyError, useApolloClient } from '../../../utils/apollo_context';
+import { useTrackedPromise } from '../../../utils/use_tracked_promise';
+import { logSummaryHighlightsQuery } from './log_summary_highlights.gql_query';
+
+export type LogSummaryHighlights = LogSummaryHighlightsQuery.Query['source']['logSummaryHighlightsBetween'];
+
+export const useLogSummaryHighlights = (
+  sourceId: string,
+  sourceVersion: string | undefined,
+  start: number | null,
+  end: number | null,
+  bucketSize: number,
+  filterQuery: string | null,
+  highlightTerms: string[]
+) => {
+  const apolloClient = useApolloClient();
+  const [logSummaryHighlights, setLogSummaryHighlights] = useState<LogSummaryHighlights>([]);
+
+  const [loadLogSummaryHighlightsRequest, loadLogSummaryHighlights] = useTrackedPromise(
+    {
+      cancelPreviousOn: 'resolution',
+      createPromise: async () => {
+        if (!apolloClient) {
+          throw new DependencyError('Failed to load source: No apollo client available.');
+        }
+        if (!start || !end || !highlightTerms.length) {
+          throw new Error();
```

Comment:
Should this error have a message? What does this throw do?

Reply:
It should have a message, even though it never bubbles up beyond the hook. It's used to short-circuit the loading mechanism. Maybe a `return Promise.reject()` would be more apppriate. :thinking: